codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -445,26 +445,24 @@ public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promi
                 throw connectionError(PROTOCOL_ERROR, "A client cannot push.");
             }
 
-            Http2Stream parentStream = connection.stream(streamId);
-
-            if (shouldIgnoreHeadersOrDataFrame(ctx, streamId, parentStream, "PUSH_PROMISE")) {
+            if (streamCreatedAfterGoAwaySent(promisedStreamId)) {
                 return;
             }
 
-            if (parentStream == null) {
-                throw connectionError(PROTOCOL_ERROR, "Stream %d does not exist", streamId);
-            }
+            Http2Stream parentStream = connection.stream(streamId);
 
-            switch (parentStream.state()) {
-              case OPEN:
-              case HALF_CLOSED_LOCAL:
-                  // Allowed to receive push promise in these states.
-                  break;
-              default:
-                  // Connection error.
-                  throw connectionError(PROTOCOL_ERROR,
-                      "Stream %d in unexpected state for receiving push promise: %s",
-                      parentStream.id(), parentStream.state());
+            if (parentStream != null) {
+                switch (parentStream.state()) {
+                    case OPEN:
+                    case HALF_CLOSED_LOCAL:
+                        // Allowed to receive push promise in these states.
+                        break;
+                    default:
```

Comment:
If the stream was reset, we could be in the CLOSED state, and this will interpret it as a connection error. I don't know if we'll still have a `Http2Stream` at that point though, so it may not be possible.

Reply:
I'm not sure it's possible either, per my understanding of the code we remove the stream from the connection when we send the RST_STREAM

Reply:
@mosesn @ejona86 can you please have a final look ....